frontend/src/metabase/components/AdminPaneLayout.jsx

```diff
@@ -12,28 +14,33 @@ const AdminPaneTitle = ({
   buttonDisabled,
   buttonLink,
   headingContent,
-}) => (
-  <section className="clearfix px2">
-    {buttonText && buttonLink && (
-      <Link to={buttonLink} className="inline-block float-right">
-        <Button primary>{buttonText}</Button>
-      </Link>
-    )}
-    {buttonText && buttonAction && (
-      <Button
-        className="float-right"
-        primary={!buttonDisabled}
-        disabled={buttonDisabled}
-        onClick={buttonAction}
-      >
-        {buttonText}
-      </Button>
-    )}
-    {headingContent && <React.Fragment>{headingContent}</React.Fragment>}
-    {title && <h2 className="PageTitle">{title}</h2>}
-    {description && <p className="text-measure">{description}</p>}
-  </section>
-);
+}) => {
+  const buttonClassName = "ml-auto flex-no-shrink";
+  return (
+    <Container>
+      <HeadingContainer>
+        {headingContent && <React.Fragment>{headingContent}</React.Fragment>}
+        {title && <h2 className="PageTitle">{title}</h2>}
+        {buttonText && buttonLink && (
+          <Link to={buttonLink} className={buttonClassName}>
+            <Button primary>{buttonText}</Button>
+          </Link>
+        )}
+        {buttonText && buttonAction && (
+          <Button
+            className={buttonClassName}
+            primary={!buttonDisabled}
+            disabled={buttonDisabled}
+            onClick={buttonAction}
+          >
+            {buttonText}
+          </Button>
+        )}
```

Comment:
Move `Link` and `Button` to be after `headingContent` and `title` as they're now parts of the same flex container as `headingContent` and `title`.

Previously they were placed at the start to avoid being on a different line as `headingContent` and `title` and `float: right` will make the button float on the right.

The new code will now look more like the actual UI.